### PR TITLE
ci: Add GitHub action to auto deploy doc build

### LINKED ISSUE #52: Add GitHub action for deploying the ComptoxAI website
We will need to create a GitHub action that deploys the website to `comptox.ai`. This action should run any time a new release of ComptoxAI is created, and should only trigger if the action for building the website is successful (see #51 for details). 

Currently, the website is hosted on a physical server on the UPenn campus, but we will be migrating to AWS in the near future. This action should target AWS rather than the physical server, so we may need to delay implementation of this action until the migration to AWS has been completed.

### PR DESCRIPTION
**Pull request by yunchae-kim**
[#52] Refer to GitHub issue…

This update makes the following changes:

- Created GitHub action to automatically deploy doc build with Sphinx
and React, and test before push

- Created GitHub action templates

- Modified conf.py and package.json files to fix dependency issues and
bugs

- Reformatted GitHub action files to standardize to format guidelines

Closes #52

### CODE REVIEW

File: .github/workflow-templates/romano-lab-ci.yml
Comment (yunchae-kim) [inline .github/workflow-templates/romano-lab-ci.yml:1]: Deleted unnecessary file.

File: .github/workflow-templates/romano-lab-ci.properties.json
Comment (yunchae-kim) [inline .github/workflow-templates/romano-lab-ci.properties.json:1]: Deleted unnecessary file.

File: .github/workflow-templates/romanolab-ci-build.yml
Comment (yunchae-kim) [inline .github/workflow-templates/romanolab-ci-build.yml:1]: Template for building doc.

File: .github/workflow-templates/romanolab-ci-python-package.yml
Comment (yunchae-kim) [inline .github/workflow-templates/romanolab-ci-python-package.yml:1]: Renamed python package test action.

File: .github/workflow-templates/romanolab-ci-python-test.yml
Comment (yunchae-kim) [inline .github/workflow-templates/romanolab-ci-python-test.yml:1]: Template for python test action.

File: .github/workflows/ci-python-package.yml
Comment (yunchae-kim) [inline .github/workflows/ci-python-package.yml:1]: Renamed python package action.

File: .github/workflows/ci-python-test.yml
Comment (yunchae-kim) [inline .github/workflows/ci-python-test.yml:1]: Renamed python test action.

File: .github/workflows/python-package.yml
Comment (yunchae-kim) [inline .github/workflows/python-package.yml:1]: Renamed.

File: docs/source/conf.py
Comment (yunchae-kim) [inline docs/source/conf.py:1]: Auto formatted conf.py file and added sphinx.ext.coverage for future use.

File: package.json
Comment (yunchae-kim) [inline package.json:1]: Deleted @babel/plugin-proposal-private-property-in-object as it is moved to package.json in app folder.

**Review by JDRomano2 — APPROVED** [review]

**Pull request merged**